addons/stock/models/stock_location.py

```diff
@@ -167,6 +167,12 @@ def _onchange_usage(self):
         if self.usage not in ('internal', 'inventory'):
             self.scrap_location = False
 
+    @api.constrains('scrap_location')
+    def _check_scrap_location(self):
+        for record in self:
+            if record.scrap_location and self.env['stock.picking.type'].search_count([('code', '=', 'mrp_operation'), '|', ('default_location_src_id', '=', record.id), ('default_location_dest_id', '=', record.id)]):
+                raise ValidationError("You cannot set a location of type Scrap, to a manufacturing type operation.")
```

Comment:
Don't forget to add _() around the string for translation purposes (same comment applies to the `_check_default_location`)